### ISSUE TITLE
Unexpected Nonprintable characters seen in some EXPLAIN plans

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#23921]
https://buildkite.com/materialize/nightlies/builds/5613#018c6b79-fb13-458e-9025-01a2c07419a5

### What version of Materialize are you using?

e3192aaebd

### What is the issue?

```
_bk;t=1702613564145testdrive/specialized_arrangements.td:227:1: error: non-matching rows: expected:

_bk;t=1702613564145[["Explained Query:\n  Join::Delta\n    plan_path[0]\n      delta_stage[1]\n        closure\n          project=(#1)\n        lookup={ relation=2, key=[#0] }\n        stream={ key=[#0], thinning=(#1) }\n      delta_stage[0]\n        closure\n          project=(#0, #2)\n        lookup={ relation=1, key=[#0, #1] }\n        stream={ key=[#0, #1], thinning=(#2) }\n      initial_closure\n        project=(#0..=#2)\n        filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n        map=((((#1 + #0) + #0) / 3))\n      source={ relation=0, key=[#0] }\n    plan_path[1]\n      delta_stage[1]\n        closure\n          project=(#1)\n        lookup={ relation=2, key=[#0] }\n        stream={ key=[#0], thinning=(#1) }\n      delta_stage[0]\n        closure\n          project=(#0, #2)\n          filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n          map=((((#1 + #0) + #0) / 3))\n        lookup={ relation=0, key=[#0, #1] }\n        stream={ key=[#0, #1], thinning=() }\n      source={ relation=1, key=[#0, #1] }\n    plan_path[2]\n      delta_stage[1]\n        closure\n          project=(#2)\n        lookup={ relation=1, key=[#0, #1] }\n        stream={ key=[#1, #0], thinning=(#2) }\n      delta_stage[0]\n        closure\n          project=(#1, #0, #2)\n          filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n          map=((((#1 + #0) + #0) / 3))\n        lookup={ relation=0, key=[#0] }\n        stream={ key=[#0], thinning=() }\n      source={ relation=2, key=[#0] }\n    Get::PassArrangements materialize.public.v_empty\n      raw=false\n      arrangements[0]={ key=[#0, #1], permutation=id, thinning=() }\n      arrangements[1]={ key=[#0], permutation=id, thinning=(#1) }\n      types=[integer?, integer?]\n    Get::PassArrangements materialize.public.t\n      raw=false\n      arrangements[0]={ key=[#0, #1], permutation=id, thinning=() }\n      types=[integer?, integer?]\n    Get::PassArrangements materialize.public.t2\n      raw=false\n      arrangements[0]={ key=[#0], permutation=id, thinning=() }\n      types=[integer?]\n\nUsed Indexes:\n  - materialize.public.v_empty_primary_idx (delta join 1st input (full scan))\n  - materialize.public.v_a (delta join 1st input (full scan))\n  - materialize.public.t2_primary_idx (delta join lookup)\n  - materialize.public.t_a_b (delta join lookup)\n"]]

_bk;t=1702613564145got:

_bk;t=1702613564145[["Explained Query:\n  Join::Delta\n    plan_path[0]\n      delta_stage[1]\n        closure\n          project=(#1)\n        lookup={ relation=2, key=[#0] }\n        stream={ key=[#0], thinning=(#1) }\n      delta_stage[0]\n        closure\n          project=(#0, #2)\n        lookup={ relation=1, key=[#0, #1] }\n        stream={ key=[#0, #1], thinning=(#2) }\n      initial_closure\n        project=(#0..=#2)\n        filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n        map=((((#1 + #0) + #0) / █))\n      source={ relation=0, key=[#0] }\n    plan_path[1]\n      delta_stage[1]\n        closure\n          project=(#1)\n        lookup={ relation=2, key=[#0] }\n        stream={ key=[#0], thinning=(#1) }\n      delta_stage[0]\n        closure\n          project=(#0, #2)\n          filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n          map=((((#1 + #0) + #0) / █))\n        lookup={ relation=0, key=[#0, #1] }\n        stream={ key=[#0, #1], thinning=() }\n      source={ relation=1, key=[#0, #1] }\n    plan_path[2]\n      delta_stage[1]\n        closure\n          project=(#2)\n        lookup={ relation=1, key=[#0, #1] }\n        stream={ key=[#1, #0], thinning=(#2) }\n      delta_stage[0]\n        closure\n          project=(#1, #0, #2)\n          filter=((#0) IS NOT NULL AND (#1) IS NOT NULL)\n          map=((((#1 + #0) + #0) / █))\n        lookup={ relation=0, key=[#0] }\n        stream={ key=[#0], thinning=() }\n      source={ relation=2, key=[#0] }\n    Get::PassArrangements materialize.public.v_empty\n      raw=false\n      arrangements[0]={ key=[#0, #1], permutation=id, thinning=() }\n      arrangements[1]={ key=[#0], permutation=id, thinning=(#1) }\n      types=[integer?, integer?]\n    Get::PassArrangements materialize.public.t\n      raw=false\n      arrangements[0]={ key=[#0, #1], permutation=id, thinning=() }\n      types=[integer?, integer?]\n    Get::PassArrangements materialize.public.t2\n      raw=false\n      arrangements[0]={ key=[#0], permutation=id, thinning=() }\n      types=[integer?]\n\nUsed Indexes:\n  - materialize.public.v_empty_primary_idx (delta join 1st input (full scan))\n  - materialize.public.v_a (delta join 1st input (full scan))\n  - materialize.public.t2_primary_idx (delta join lookup)\n  - materialize.public.t_a_b (delta join lookup)\n"]]
```

and

```
_bk;t=1702612958562testdrive/monotonic.td:79:1: error: non-matching rows: expected:

_bk;t=1702612958562[["Explained Query:\n  TopK::MonotonicTopK order_by=[#0 asc nulls_last] limit=3 must_consolidate\n    ArrangeBy\n      input_key=[#0]\n      raw=true\n      Get::PassArrangements materialize.public.i1\n        raw=false\n        arrangements[0]={ key=[#0], permutation=id, thinning=() }\n        types=[bigint]\n\nUsed Indexes:\n  - materialize.public.i1_primary_idx (*** full scan ***)\n"]]

_bk;t=1702612958562got:

_bk;t=1702612958562[["Explained Query:\n  TopK::MonotonicTopK order_by=[#0 asc nulls_last] limit=█ must_consolidate\n    ArrangeBy\n      input_key=[#0]\n      raw=true\n      Get::PassArrangements materialize.public.i1\n        raw=false\n        arrangements[0]={ key=[#0], permutation=id, thinning=() }\n        types=[bigint]\n\nUsed Indexes:\n  - materialize.public.i1_primary_idx (*** full scan ***)\n"]]
```

The `limit` in the second failure is an example of a non-printable character

**Comment by philip-stoev** [comment]
To reproduce:

```
test/cloudtest/setup
test/cloudtest/pytest -mlong test/cloudtest/test_full_testdrive.py
```

This will run all the tests, so will take a while. In order to jump directly to an offending test:

1. Kill the above command with Ctrl+C,
2. Run directly:
```
kubectl --context kind-mzcloud --namespace default exec -it testdrive -- testdrive --materialize-url=postgres://materialize:materialize@environmentd:6875/materialize --materialize-internal-url=postgres://mz_system@environmentd:6877/materialize --kafka-addr=redpanda:9092 --schema-registry-url=http://redpanda:8081 --default-timeout=300s --log-filter=off --var=replicas=1 --var=default-storage-size=1 --var=default-replica-size=1 testdrive/monotonic.td testdrive/specialized_arrangements.td
```

**Comment by philip-stoev** [comment]
Not reproducible with `v0.79.1`

**Comment by aalexandrov** [comment]
This is expected for plans that run against containers that don't have soft assertions enabled.

My `EXPLAIN WITH(redacted)` PR added the following lines:

```

```

Change them to `false` if you can't control the soft assertions flag for all CI runs.

**Comment by aalexandrov** [comment]
@philip-stoev is this still an issue? I thought it was resolved by #23924?

**Comment by aalexandrov** [comment]
Closing this after an offline discussion with @philip-stoev as this is no longer a problem for CI tests.
